### PR TITLE
Fix 'do not assign a lambda expression, use a def'

### PR DESCRIPTION
**Pull request by rosswhitfield**
This should fix all flake8 E731 warnings.

**To test:**
- Code review

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/isis_reduction_steps.py [hunk 1]
```diff
@@ -1843,7 +1846,9 @@ def _get_final_setting_monitors(self, run_number, use_mean, use_time, mon_number
                     monitor_mon_numbers.append(mon_numbers[index])
 
         # Check if the mean value is identical for all entries
-        are_all_same = lambda a_list : all([a_list[0] == a_list[i] for i in range(0, len(a_list))])
+        def are_all_same(a_list):
```
Comment (AndreiSavici) [inline scripts/SANS/isis_reduction_steps.py]: It is used only once. Try instead `len(set(a_list))==1`

File: scripts/SANS/isis_reduction_steps.py [hunk 2]
```diff
@@ -1753,15 +1753,18 @@ def _evaluate_settings(self):
             run_number.append(setting.run_number)
 
         # Get the indices with settings which correspond to the individual settings
-        get_indices = lambda time_flag, mon_flag : [i for i, val in enumerate(use_time)
-                                                    if use_time[i] == time_flag and use_mon[i] == mon_flag]
+        def get_indices(time_flag, mon_flag):
+            return [i for i, val in enumerate(use_time) if use_time[i] == time_flag and use_mon[i] == mon_flag]
```
Comment (AndreiSavici) [inline scripts/SANS/isis_reduction_steps.py]: Are you using val anywhere?

Comment (rosswhitfield) [inline scripts/SANS/isis_reduction_steps.py]: No, doesn't look like it.

Comment (rosswhitfield) [inline scripts/SANS/isis_reduction_steps.py]: Although `use_time[i]` can be replaced with `val`

**Comment by AndreiSavici** [comment]
LGTM :shipit:

**Pull request merged**